initialize.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+set -e
+
+# Check if docker is installed
+if ! [ -x "$(command -v docker)" ]; then
+  echo 'Error: docker is not installed.' >&2
+  # Ask if user wants to install docker
+  read -p "Do you want to install docker? [y/n] " -n 1 -r
+  echo
+  if [[ $REPLY =~ ^[Yy]$ ]]; then
+    # Install docker
+    curl -fsSL https://get.docker.com -o get-docker.sh
+    sh get-docker.sh
+    # Check if docker is installed
+    if ! [ -x "$(command -v docker)" ]; then
+      echo 'Error: Could not install docker.' >&2
+      exit 1
+    fi
+  else
+    exit 1
+  fi
+fi
+
+# Check if docker-compose is installed
+if ! [ -x "$(command -v docker-compose)" ]; then
+  echo 'Error: docker-compose is not installed.' >&2
+  # Ask if user wants to install docker-compose
+  read -p "Do you want to install docker-compose? [y/n] " -n 1 -r
+  echo
+  if [[ $REPLY =~ ^[Yy]$ ]]; then
+    # Install docker-compose
+    sudo curl -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+    sudo chmod +x /usr/local/bin/docker-compose
+    sudo ln -s /usr/local/bin/docker-compose /usr/bin/docker-compose
+    # Check if docker-compose is installed
+    if ! [ -x "$(command -v docker)" ]; then
+      echo 'Error: Could not install docker-compose.' >&2
+      exit 1
+    fi
+  else
+    exit 1
+  fi
+fi
+
+# Check if python is installed
+if ! [ -x "$(command -v python)" ]; then
```

Comment:
I would look for `python3` instead. Plus could you add the installation too? See: https://linuxize.com/post/how-to-install-python-3-9-on-ubuntu-20-04/

Then, if python is already installed we would need to check if the requirement is satisfied (v3.6+)

Reply:
If the version is old, we won't manage this problem because too hard. But we need to tell them that they need to change the version so the users can do on their own

Reply:
> I would look for `python3` instead. Plus could you add the installation too? See: https://linuxize.com/post/how-to-install-python-3-9-on-ubuntu-20-04/
> 
> Then, if python is already installed we would need to check if the requirement is satisfied (v3.6+)

@mlodic I thought of doing this. But that would limit us to systems using apt. The other scripts (docker and docker-compose) I've used support a wider variety of distros

Reply:
this makes sense, thank you for the explanation. so I think it is enough to just tell it to the user